src/js/__tests__/components/demo/DemoForm.test.tsx

```diff
@@ -230,7 +231,7 @@ describe('DemoForm', () => {
     it('renders a form that can not be submitted while values are missing', () => {
         const submitButton = screen.getByText('Submit Form');
 
-        fireEvent.click(submitButton);
+        fireEvent.submit(submitButton);
```

Comment:
Now this is a weird issue.

Apparently fireEvent.click does not fire the `onSubmit` function of forms when running in vitest but does so in jest. I found a couple of stray StackOverflow questions that seem to run into the same issue and there hasn't been any solution besides switching from `click` to `submit`. I don't particularly like that this change is necessary, but I think I don't see a lot of value in trying to debug this further - if you disagree and *would* prefer staying with `click`, I can spend some more time trying to find a better solution though!